bagua/torch_api/algorithms/decentralized.py

```diff
@@ -68,7 +66,85 @@ def init_operations(
     ):
         bucket.clear_ops()
         bucket.append_decentralized_synchronous_op(
-            hierarchical=True,
+            hierarchical=self.hierarchical,
             peer_selection_mode=self.peer_selection_mode,
             communication_interval=self.communication_interval,
         )
+
+
+class LowPrecisionDecentralizedAlgorithm(Algorithm):
+    def __init__(self, hierarchical: bool = True, communication_interval: int = 1):
+        """
+        Create an instance of the
+        `Difference Compression Decentralized <https://arxiv.org/pdf/1803.06443.pdf>`_
+        algorithm.
+
+        Args:
+            hierarchical (bool): Enable hierarchical communication.
+            communication_interval (int): Number of iterations between two communication steps.
+        """
+        self.hierarchical = hierarchical
+        self.communication_interval = communication_interval
+
+    def init_tensors(self, bagua_module: BaguaModule) -> List[BaguaTensor]:
+        parameters = bagua_module.bagua_build_params()
+        self.tensors = [
+            param.ensure_bagua_tensor(name) for name, param in parameters.__reversed__()
```

Comment:
need sanity check that the set of bagua tensors are the same as the set of tensors in optimizer param_group so that training won't hang at `post_optimizer_step_hook`.

Reply:
slow down down, I am not done yet..

Reply:
sure, mark the pr as ready for review when ready